x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/evaluator/predicate/operator/comparison/InsensitiveEquals.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.esql.evaluator.predicate.operator.comparison;
+
+import org.apache.lucene.util.BytesRef;
+import org.apache.lucene.util.automaton.ByteRunAutomaton;
+import org.elasticsearch.compute.ann.Evaluator;
+import org.elasticsearch.compute.ann.Fixed;
+import org.elasticsearch.xpack.ql.expression.Expression;
+import org.elasticsearch.xpack.ql.tree.NodeInfo;
+import org.elasticsearch.xpack.ql.tree.Source;
+import org.elasticsearch.xpack.ql.type.DataTypes;
+
+import java.time.ZoneId;
+
+public class InsensitiveEquals extends InsensitiveBinaryComparison {
+
+    public InsensitiveEquals(Source source, Expression left, Expression right, ZoneId zoneId) {
+        super(source, left, right, zoneId);
+    }
+
+    @Override
+    public InsensitiveBinaryComparison reverse() {
+        return this;
+    }
+
+    @Override
+    protected NodeInfo<InsensitiveEquals> info() {
+        return NodeInfo.create(this, InsensitiveEquals::new, left(), right(), zoneId());
+    }
+
+    @Override
+    protected InsensitiveEquals replaceChildren(Expression newLeft, Expression newRight) {
+        return new InsensitiveEquals(source(), newLeft, newRight, zoneId());
+    }
+
+    @Evaluator(extraName = "Ints")
+    static boolean processInts(int lhs, int rhs) {
+        return lhs == rhs;
+    }
+
+    @Evaluator(extraName = "Longs")
+    static boolean processLongs(long lhs, long rhs) {
+        return lhs == rhs;
+    }
+
+    @Evaluator(extraName = "Doubles")
+    static boolean processDoubles(double lhs, double rhs) {
+        return lhs == rhs;
+    }
+
+    @Evaluator(extraName = "Keywords")
+    static boolean processKeywords(BytesRef lhs, BytesRef rhs) {
+        return lhs.utf8ToString().equalsIgnoreCase(rhs.utf8ToString());
```

Comment:
I don't believe this'll work for `BytesRef`. I think that supports non-utf8 text.

I'm also not 100% confident `equalsIgnoreCase` will have the same behavior as the Lucene queries that come out of the field mappers, though I haven't checked. if you have then cool.